### PR TITLE
ENH: support no images for MLX vlm

### PR DESCRIPTION
**Pull request by qinxuye**
Before, VLM model must have images for MLX-vlm. This PR addresses this issue.

image

### CODE REVIEW

File: xinference/model/llm/mlx/core.py [hunk 1]
```diff
@@ -471,13 +477,47 @@ def load(self):
...
+            def __init__(self, model):
+                super().__init__()
+                self._model = model.language_model
+
+            @property
+            def layers(self):
+                return self._model.layers
+
+            def __call__(self, *args, **kwargs):
+                return self._model(*args, **kwargs).logits
```
[the remark this answers is not in the document]
Comment (qinxuye) [reply xinference/model/llm/mlx/core.py:494]: The  language model in mlx-vlm is different with the model in mlx-lm, the later on will return logits, but language model in vlm returns https://github.com/Blaizzy/mlx-vlm/blob/3f5e1620072440afb7496940f67ac1c7fc64056f/mlx_vlm/models/base.py#L260

This will cause failure in stream_generate.

[the remark this answers is not in the document]
Comment (qinxuye) [reply xinference/model/llm/mlx/core.py:494]: I will add a comment to explain it.

File: xinference/model/llm/mlx/core.py [hunk 2]
```diff
@@ -471,13 +477,52 @@ def load(self):
...
+
+        sampler = make_sampler(
+            temp=kwargs.pop("temperature"), top_p=kwargs.pop("top_p")
+        )
```
[the remark this answers is not in the document]
Comment (qinxuye) [reply xinference/model/llm/mlx/core.py:503]: There is default value for temperature.
